src/components/ShareMyLocationButton.js

```diff
@@ -19,41 +19,30 @@ const ShareMyLocationButton = () => {
     const newMapCenter = { lat, lng };
     setUserCenterMap(newMapCenter);
 
-    if (!userLocationShared) {
+    if (!sharingLocation.current) {
       setNewCenterMap(newMapCenter);
-      setUserLocationShared(true);
+      sharingLocation.current = true;
+      setUserLocationShared(sharingLocation.current);
     }
   };
   const handleError = () => setUserCenterMap(defaultCenterMap);
 
-  const buttonText = userLocationShared
-    ? 'Stop Sharing Location'
-    : 'Share My Location';
-
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      const trackingId = navigator.geolocation.watchPosition(
-        handleSuccess,
-        handleError,
-      );
-
-      setTrackingId(trackingId);
-    }
-  };
-
-  const handleStopSharing = () => {
-    if (userLocationShared && trackingId) {
-      setTrackingId(navigator.geolocation.clearWatch(trackingId));
+      if (trackingId === null) {
+        const trackingId = navigator.geolocation.watchPosition(
+          handleSuccess,
+          handleError,
+        );
+
+        setTrackingId(trackingId);
+      }
     }
-    setUserLocationShared(false);
-    setNewCenterMap(defaultCenterMap);
   };
 
   return (
-    <button
```

Comment:
We'll also need to preserve these changes, including the `buttonText`. Once the user shares their location, the button text should change to "Stop Sharing Location", and the `onClick` function to change to `handleStopSharing`.